libhirte/src/service/shutdown.c

```diff
@@ -24,18 +25,49 @@ static int method_shutdown(sd_bus_message *m, void *userdata, UNUSED sd_bus_erro
         return sd_bus_reply_method_return(m, "");
 }
 
+#define method_name_shutdown "Shutdown"
+
 // NOLINTNEXTLINE(cppcoreguidelines-interfaces-global-init)
-static const sd_bus_vtable vtable_shutdown[] = { SD_BUS_VTABLE_START(0),
-                                                 SD_BUS_METHOD("Shutdown", "", "", method_shutdown, 0),
-                                                 SD_BUS_VTABLE_END };
+static const sd_bus_vtable vtable_shutdown[] = {
+        SD_BUS_VTABLE_START(0), SD_BUS_METHOD(method_name_shutdown, "", "", method_shutdown, 0), SD_BUS_VTABLE_END
+};
 
 bool service_register_shutdown(sd_bus *target_bus, sd_event *event_loop) {
-        _cleanup_free_ char *interface_name = assemble_interface_name("Shutdown");
+        _cleanup_free_ char *interface_name = assemble_interface_name(method_name_shutdown);
         int r = sd_bus_add_object_vtable(
                         target_bus, NULL, HIRTE_OBJECT_PATH, interface_name, vtable_shutdown, event_loop);
         if (r < 0) {
                 fprintf(stderr, "Failed to register shutdown service: %s\n", strerror(-r));
                 return false;
         }
         return true;
-}
\ No newline at end of file
+}
+
+bool service_call_shutdown(sd_bus *target_bus, const char *service_name) {
```

Comment:
The more usual C style here is to return an int that can contain either -1 or some negative value providing an error code.  Boolean return values should be used only for very simple functions for which an error has no value.

Reply:
Yes, as we discussed yesterday, we need to refactor this - maybe introduce some custom error codes where it makes sense. This has to be a separate task, though, as the changes are affecting a lot of code.

Reply:
Refactored this part accordingly so that it can serve as a reference.